streams/src/test/java/org/apache/kafka/streams/integration/StreamsResetterIntegrationTest.java

```diff
@@ -0,0 +1,811 @@
+            resultTopicConsumerConfig,
+            OUTPUT_TOPIC_2_RERUN,
+            40);
+        streams.close();
+
+        assertThat(resultRerun, equalTo(result));
+        assertThat(resultRerun2, equalTo(result2));
+
+        TestUtils.waitForCondition(consumerGroupInactive, TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT,
+            "Reset Tool consumer group did not time out after " + (TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT) + " ms.");
+        cleanGlobal(null, null);
+
+        cluster.deleteTopicAndWait(INTERMEDIATE_USER_TOPIC);
+    }
+
+    @Test
+    public void testReprocessingFromLatestAfterResetWithoutIntermediateUserTopic() throws Exception {
+        final Properties streamsConfiguration = prepareTest();
+
+        final Properties resultTopicConsumerConfig = new Properties();
+        resultTopicConsumerConfig.putAll(TestUtils.consumerConfig(
+            bootstrapServers,
+            APP_ID + "-standard-consumer-" + OUTPUT_TOPIC,
+            LongDeserializer.class,
+            LongDeserializer.class));
+
+        // RUN
+        KafkaStreams streams = new KafkaStreams(setupTopologyWithoutIntermediateUserTopic(), streamsConfiguration);
+        streams.start();
+        final List<KeyValue<Long, Long>> result = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(
+            resultTopicConsumerConfig,
+            OUTPUT_TOPIC,
+            10);
+
+        streams.close();
+        TestUtils.waitForCondition(consumerGroupInactive, TIMEOUT_MULTIPLIER * STREAMS_CONSUMER_TIMEOUT,
+            "Streams Application consumer group did not time out after " + (TIMEOUT_MULTIPLIER * STREAMS_CONSUMER_TIMEOUT) + " ms.");
+
+        // RESET
+        streams = new KafkaStreams(setupTopologyWithoutIntermediateUserTopic(), streamsConfiguration);
+        streams.cleanUp();
+
+        cleanGlobal("--to-latest", null);
+        TestUtils.waitForCondition(consumerGroupInactive, TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT,
+            "Reset Tool consumer group did not time out after " + (TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT) + " ms.");
+
+        //ADD 10 MORE
+        add10InputElements();
+
+        assertInternalTopicsGotDeleted(null);
+
+        // RE-RUN
+        streams.start();
+        final List<KeyValue<Long, Long>> resultRerun = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(
+            resultTopicConsumerConfig,
+            OUTPUT_TOPIC,
+            10);
+        streams.close();
+
+        assertThat(resultRerun, equalTo(result));
+
+        TestUtils.waitForCondition(consumerGroupInactive, TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT,
+            "Reset Tool consumer group did not time out after " + (TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT) + " ms.");
+        cleanGlobal(null, null);
+    }
+
+    @Test
+    public void testReprocessingFromOffsetAfterResetWithoutIntermediateUserTopic() throws Exception {
```

Comment:
@jeqo Maybe my previous comment was misleading :) I was suggesting that we do not need to add more test cases into the reset integration tests like "from offset", "from date" etc, as for integration test we just check its interaction with the cluster module all worked as expected, so having just one test for "from scratch with intermediate" and "from scratch without intermediate" is good enough. For the added functionality our added unit tests should be well covering it.

Also it seems you have removed the test with SSL config as well? Is that intentional? If not maybe we can just make this integration test a parameterized test, with ssl turned on and off (i.e. when it is turned on broker cluster will have the SSL configs to not accept unauthorized access, and the streams resetter needs the appropriate SSL configs to be able to delete those topics).

cc @mjsax

Reply:
I think I get it know: I will return tests with SSL config (I thought that was what it was already tested) and remove tests that are already validated by unit-test. There are some methods from `KafkaConsumer` that are not available on unit test like `offsetsForTimes`, I will keep those on integration test.

Reply:
Sounds good! Thanks. The reason we'd like to avoid redundant integration tests with overlapping unit tests is that they would take much more time. If we are not careful the Jenkins builds would take longer and longer to finish, and hence hinders our development pace (we have seen the Streams unit test suite taking from 1 min to about 7 min now on my laptop, and much longer on the Jenkins box).

Reply:
Make total sense 👍